static/app/components/events/interfaces/frame/stacktraceLink.tsx

```diff
@@ -1,207 +1,185 @@
 import StacktraceLinkModal from './stacktraceLinkModal';
 
-type Props = AsyncComponent['props'] & {
+interface StacktraceLinkProps {
   event: Event;
   frame: Frame;
+  /**
+   * The line of code being linked
+   */
   line: string;
-  lineNo: number;
-  organization: Organization;
-  projects: Project[];
-};
-
-type State = AsyncComponent['state'] & {
-  isDismissed: boolean;
-  match: StacktraceLinkResult;
-  promptLoaded: boolean;
-};
+}
 
-class StacktraceLink extends AsyncComponent<Props, State> {
-  get project() {
-    // we can't use the withProject HoC on an the issue page
-    // so we ge around that by using the withProjects HoC
-    // and look up the project from the list
-    const {projects, event} = this.props;
-    return projects.find(project => project.id === event.projectID);
-  }
+export function StacktraceLink({frame, event, line}: StacktraceLinkProps) {
+  const api = useApi();
+  const organization = useOrganization();
+  const {projects} = useProjects();
+  const project = useMemo(
+    () => projects.find(p => p.id === event.projectID),
+    [projects, event]
+  );
+  const [promptDismissed, setPromptDismissed] = useState(false);
+  const prompt = usePromptsCheck('stacktrace_link', organization.id, project?.id);
+
+  const query = {
+    file: frame.filename,
+    platform: event.platform,
+    commitId: event.release?.lastCommit?.id,
+    ...(event.sdk?.name && {sdkName: event.sdk.name}),
+    ...(frame.absPath && {absPath: frame.absPath}),
+    ...(frame.module && {module: frame.module}),
+    ...(frame.package && {package: frame.package}),
+  };
+  const {
+    data: match,
+    isLoading,
+    refetch,
+  } = useQuery<StacktraceLinkResult>([
+    `/projects/${organization.slug}/${project?.slug}/stacktrace-link/`,
+    {query},
+  ]);
+
+  // Update dismissed state when prompt is loaded
+  useEffect(() => {
+    if (prompt.isLoading || !prompt.data?.data) {
+      return;
+    }
 
-  componentDidMount() {
-    this.promptsCheck();
-  }
+    setPromptDismissed(
+      promptIsDismissed({
+        dismissedTime: prompt.data.data.dismissed_ts,
+        snoozedTime: prompt.data.data.snoozed_ts,
+      })
+    );
+  }, [prompt.isLoading, prompt.data]);
 
-  async promptsCheck() {
-    const {organization} = this.props;
+  // Track stacktrace analytics after loaded
+  useEffect(() => {
+    if (isLoading || !match) {
+      return;
+    }
 
-    const prompt = await promptsCheck(this.api, {
-      organizationId: organization.id,
-      projectId: this.project?.id,
-      feature: 'stacktrace_link',
+    trackIntegrationAnalytics('integrations.stacktrace_link_viewed', {
+      view: 'stacktrace_issue_details',
+      organization,
+      platform: project?.platform,
+      project_id: project?.id,
+      state:
+        // Should follow the same logic in render
+        match.sourceUrl
+          ? 'match'
+          : match.error || match.integrations.length > 0
+          ? 'no_match'
+          : !promptDismissed
```

Comment:
This seems a bit weird since the dismissed query might not complete before this gets triggered